client/src/pages/Discovery/Discovery.tsx

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useState } from 'react';
+import Grow from '@material-ui/core/Grow';
+import {
+  Avatar,
+  Box,
+  Button,
+  Card,
+  CardActionArea,
+  CardActions,
+  CardContent,
+  Grid,
+  Typography,
+} from '@material-ui/core';
+import useStyles from './useStyles';
+import { Contest } from '../../interface/Contest';
+import { getAllContests } from '../../helpers/APICalls/contest';
+
+export default function Discovery(): JSX.Element {
+  const classes = useStyles();
+  const [contestObj, setContestObj] = useState<Contest[]>();
+
+  useEffect(() => {
+    getAllContests().then((data) => {
```

Comment:
Maybe check here for errors from the response as well?

Reply:
added error handling